### PR TITLE
@benisgold/fix polygonscan error

### PR DESCRIPTION
**Pull request by benisgold**
Fixes RNBW-2350

## What changed (plus any additional context for devs)
-  Polygonscan link was broken for some L2 tokens because `assetWithPrice.l2Address` is sometimes undefined. The address that gets passed into the polygonscan url now falls back to `asset?.address` if the former is undefined.
- Changed "Polygon Scan" to "Polygonscan"
- Removed random apostrophe that appears in the discover page token info boxes

## PoW (screenshots / screen recordings)
https://user-images.githubusercontent.com/15272675/152582850-d72f177a-bd73-4e6d-91e9-421580f14843.mp4

## Dev checklist for QA: what to test
Check that Polygonscan/Etherscan/etc links aren't broken

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?

### CODE REVIEW

File: src/components/ContactRowInfoButton.js [hunk 1]
```diff
@@ -97,7 +98,7 @@ const ContactRowInfoButton = ({ children, item, network, scaleTo }) => {
   );
 
   const onPressAndroid = useCallback(() => {
-    const blockExplorerText = `View on ' ${startCase(
```
Comment (skylarbarrera) [inline src/components/ContactRowInfoButton.js:100]: 😅

**Review by skylarbarrera — APPROVED** [review]
🙆‍♂️

File: src/components/ContactRowInfoButton.js (hunk 1 above)
Comment (benisgold) [reply src/components/ContactRowInfoButton.js:100]: 😤

**Review by jinchung — APPROVED** [review]
🌮

**Pull request merged**